AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvSniffer.java

```diff
@@ -0,0 +1,446 @@
+/*
+ Copyright (c) 2020 David Allison <davidallisongithub@gmail.com>
+
+ This program is free software; you can redistribute it and/or modify it under
+ the terms of the GNU General Public License as published by the Free Software
+ Foundation; either version 3 of the License, or (at your option) any later
+ version.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along with
+ this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ This file incorporates work covered by the following copyright and permission notice.
+ Please see LICENSE for full details
+
+ Ported from https://github.com/python/cpython/blob/a74eea238f5baba15797e2e8b570d153bc8690a7/Lib/csv.py#L159
```

Comment:
```suggestion
 Ported from https://github.com/python/cpython/blob/a74eea238f5baba15797e2e8b570d153bc8690a7/Lib/csv.py#L159

  This file incorporates work covered by the following copyright and permission notice.
 Please see the file LICENSE in this directory for full details

```

I read the LICENSE file, and I like this header but I'd put the ported file link above it, then an explicit reference to the LICENSE file in the directory so it is unambiguous

Further thoughts?

Needs to be in all the files in the python dir though I think?